### PR TITLE
Remove support for basic authentication

### PR DESCRIPTION
**Pull request by enj**
This change removes support for basic authn in v1.19 via the `--basic-auth-file` flag.  This functionality was deprecated in v1.16 in response to [ATR-K8S-002: Non-constant time password comparison](https://github.com/kubernetes/kubernetes/issues/81126).

Similar functionality is available via the `--token-auth-file` flag for development purposes.

Signed-off-by: Monis Khan <mok@vmware.com>

@kubernetes/sig-auth-pr-reviews 
/milestone v1.19
/priority important-longterm
/kind cleanup
/kind deprecation

xref: #81126
xref: #81152
xref: https://github.com/kubernetes/website/pull/19630

```release-note
Action Required: Support for basic authentication via the --basic-auth-file flag has been removed.  Users should migrate to --token-auth-file for similar functionality.
```

```docs
Removed "Static Password File" section from https://kubernetes.io/docs/reference/access-authn-authz/authentication/#static-password-file
https://github.com/kubernetes/website/pull/19630
```

### CODE REVIEW

**Comment by enj** [comment]
I am wondering long term what we do about basic auth related functionality in `client-go` / `kubectl` such as `NewBasicAuthRoundTripper` and `HasBasicAuth`.

File: staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go
Comment (mikedanese) [inline staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go]: This is a nice side effect.

**Comment by mikedanese** [comment]
Deprecation was in 1.16: https://github.com/kubernetes/kubernetes/commit/3d2bc6f6ae691d405e8d6bfce9d66af816454ff0

It's time. Let's hold for a few days to allow comments.

/lgtm
/approve
/hold

File: cluster/gce/gci/configure-kubeapiserver.sh
Comment (liggitt) [inline cluster/gce/gci/configure-kubeapiserver.sh]: there's a lot of plumbing to create this file and manage KUBE_PASSWORD... not sure how much should be cleaned up in this PR or not (e.g. in get-kubeconfig-basicauth, create-kubeconfig, get-kubeconfig-user-basicauth, gen-kube-basicauth, load-or-gen-kube-basicauth, create-master-auth, etc)

Comment (enj) [reply cluster/gce/gci/configure-kubeapiserver.sh]: @bentheelder @mikedanese @spiffxp @dims you guys are listed as approvers for the `cluster` dir - do you have an opinion on this?  I try to stay as far away from BASH as possible, but if y'all think it is worthwhile to clean it up, I can look into it.

Comment (BenTheElder) [reply cluster/gce/gci/configure-kubeapiserver.sh]: @cheftako for GCI changes.

I would keep to minimum changes in cluster/ generally, if we decide to clean this up at some point it'll be a deep scrub 🙃

Comment (liggitt) [reply cluster/gce/gci/configure-kubeapiserver.sh]: On reflection, I agree

**Review by liggitt — APPROVED** [review]
/hold cancel

**Comment by mikedanese** [comment]
sig-auth: Reach out to major deployers. Do a sweep of github to check for obvious dependencies.

**Comment by liggitt** [comment]
something like https://grep.app/search?q=basic-auth-file might be helpful for finding users

**Comment by enj** [comment]
Some fun history:

#6508
#7383
#7425
#26352
#20666

**Pull request merged**
